File: app/src/main/java/com/github/pockethub/android/ui/item/dialog/RefDialogItem.java
package com.github.pockethub.android.ui.item.dialog;

import android.support.annotation.NonNull;
import android.view.View;
import android.widget.RadioButton;
import android.widget.TextView;

import com.github.pockethub.android.R;
import com.github.pockethub.android.core.ref.RefUtils;
import com.github.pockethub.android.ui.item.BaseDataItem;
import com.github.pockethub.android.ui.item.BaseViewHolder;
import com.meisolsson.githubsdk.model.git.GitReference;

import butterknife.BindView;

public class RefDialogItem extends BaseDataItem<GitReference, RefDialogItem.ViewHolder> {

    private int selected;

    public RefDialogItem(GitReference dataItem, int selected) {
        super(null, dataItem, dataItem.ref().hashCode());
        this.selected = selected;
    }

    @Override
    public int getLayout() {
        return R.layout.ref_item;
    }

    @Override
    public void bind(@NonNull ViewHolder holder, int position) {
        if (RefUtils.isTag(getData())) {
            holder.refIcon.setText(R.string.icon_tag);
        } else {
            holder.refIcon.setText(R.string.icon_fork);
        }
        holder.ref.setText(RefUtils.getName(getData()));
        holder.selected.setChecked(selected == position);
    }

    @NonNull
    @Override
    public ViewHolder createViewHolder(@NonNull View itemView) {
        return new ViewHolder(itemView);
    }

    class ViewHolder extends BaseViewHolder {
        @BindView(R.id.tv_ref_icon) TextView refIcon;
        @BindView(R.id.tv_ref) TextView ref;
        @BindView(R.id.rb_selected) RadioButton selected;

        public ViewHolder(@NonNull View rootView) {
            super(rootView);
        }
    }
}

File: app/src/main/java/com/github/pockethub/android/ui/item/dialog/RefDialogItem.kt
package com.github.pockethub.android.ui.item.dialog

import com.github.pockethub.android.R
import com.github.pockethub.android.core.ref.RefUtils
import com.meisolsson.githubsdk.model.git.GitReference
import com.xwray.groupie.kotlinandroidextensions.Item
import com.xwray.groupie.kotlinandroidextensions.ViewHolder
import kotlinx.android.synthetic.main.ref_item.*

class RefDialogItem(
        val gitReference: GitReference,
        private val selected: Int
) : Item(gitReference.ref()!!.hashCode().toLong()) {

    override fun getLayout() = R.layout.ref_item

    override fun bind(holder: ViewHolder, position: Int) {
        if (RefUtils.isTag(gitReference)) {
            holder.tv_ref_icon.setText(R.string.icon_tag)
        } else {
            holder.tv_ref_icon.setText(R.string.icon_fork)
        }
        holder.tv_ref.text = RefUtils.getName(gitReference)
        holder.rb_selected.isChecked = selected == position
    }
}
